lib/context.js

```diff
@@ -52,42 +60,65 @@ module.exports = async function getContext() {
-                encoding: 'utf-8',
-              }).trim(),
-              // Get the unix timestamp of the commit then convert it to
-              // an ISO8601 UTC string
-              //
-              // https://git-scm.com/docs/pretty-formats#Documentation/pretty-formats.txt-emctem
-              // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
-              timestamp: new Date(
-                execSync('git show -s --format=%ct $BITBUCKET_COMMIT', {
-                  env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
-                  encoding: 'utf-8',
-                }).trim() * 1000
-              ).toISOString(),
-              author: {
-                name: actor,
+      // BITBUCKET_PR_ID only set on pull request triggered builds
+      // this is how we determine if it's a `push` or `pull_request`
+      const eventName = process.env.BITBUCKET_PR_ID ? 'pull_request' : 'push';
+
+      switch (eventName) {
+        case 'push':
+          return {
+            ...baseContext,
+            eventName,
+            payload: {
+              repository: {
+                name: process.env.BITBUCKET_REPO_SLUG,
+                // TODO we should remove this in a future release
+                organization: process.env.BITBUCKET_WORKSPACE,
+              },
+              organization: {
+                login: process.env.BITBUCKET_WORKSPACE,
               },
+              commits: [
+                {
+                  message: execSync('git log --format=%B -n 1 $BITBUCKET_COMMIT', {
+                    env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
+                    encoding: 'utf-8',
+                  }).trim(),
+                  // Get the unix timestamp of the commit then convert it to
+                  // an ISO8601 UTC string
+                  //
+                  // https://git-scm.com/docs/pretty-formats#Documentation/pretty-formats.txt-emctem
+                  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
+                  timestamp: new Date(
+                    execSync('git show -s --format=%ct $BITBUCKET_COMMIT', {
+                      env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
+                      encoding: 'utf-8',
+                    }).trim() * 1000
+                  ).toISOString(),
+                  author: {
+                    name: actor,
+                  },
+                },
+              ],
             },
-          ],
-        },
-      };
+          };
+        case 'pull_request':
+          return {
+            ...baseContext,
+            eventName,
+            payload: {
+              number: process.env.BITBUCKET_PR_ID,
+              repository: {
+                name: process.env.BITBUCKET_REPO_SLUG,
+              },
+              organization: {
+                login: process.env.BITBUCKET_WORKSPACE,
```

Comment:
in the near future: should we be storing the workspace UUIDs and repo UUIDs for BB (and IDs for GH)?